File: src/main/java/ch/unibas/dmi/dbis/vrem/model/collection/ArtCollection.java
package ch.unibas.dmi.dbis.vrem.model.collection;


import ch.unibas.dmi.dbis.vrem.model.exhibition.Exhibit;
import java.util.ArrayList;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import org.bson.types.ObjectId;

public class ArtCollection {

    public final String id;
    public String name;
    public final Map<String, String> metadata = new HashMap<>();
    public List<Exhibit> exhibits;

    public ArtCollection(ObjectId id, String name, List<Exhibit> exhibits) {
        this.id = id.toHexString();
        this.name = name;
        this.exhibits = exhibits;
    }

    public ArtCollection(String name, List<Exhibit> exhibits) {
        this(new ObjectId(), name, exhibits);
    }

    public List<Exhibit> getExhibits() {
        if (exhibits == null) {
            return Collections.unmodifiableList(new ArrayList<>());
        }
        return Collections.unmodifiableList(this.exhibits);
    }

}

File: src/main/kotlin/ch/unibas/dmi/dbis/vrem/model/collection/ArtCollection.kt
package ch.unibas.dmi.dbis.vrem.model.collection

import ch.unibas.dmi.dbis.vrem.model.exhibition.Exhibit
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import org.litote.kmongo.newId

/**
 * Art collection object.
 *
 * @property id The ID of the art collection (as stored in MongoDB).
 * @property name The name of the collection.
 * @property exhibits The exhibits present in the collection.
 * @property metadata Mapping of metadata attributes to descriptions for the collection.
 */
@Serializable
data class ArtCollection(
    @SerialName("_id")
    val id: String = newId<ArtCollection>().toString(),
    val name: String,
    val exhibits: List<Exhibit> = mutableListOf(),
    val metadata: Map<String, String> = mutableMapOf()
)
